packages/react/test/createRenderer.ts

```diff
@@ -23,13 +30,41 @@ export function createRenderer(globalOptions?: CreateRendererOptions): BaseUITes
   const createRendererResult = sharedCreateRenderer(globalOptions);
   const { render: originalRender } = createRendererResult;
 
-  const render = async (element: React.ReactElement<DataAttributes>, options?: RenderOptions) =>
-    act(async () => {
+  const render = async (element: React.ReactElement<DataAttributes>, options?: RenderOptions) => {
+    const renderResult = await act(async () => {
       const result = await originalRender(element, options);
       await flushMicrotasks();
       return result;
     });
 
+    if (!isJSDOM) {
+      const { userEvent: vitestUserEvent } = await import('@vitest/browser/context');
```

Comment:
This must be imported dynamically as the static import fails when executed outside of the browser environment.